v1/transform.go

```diff
@@ -0,0 +1,344 @@
+	for i, r := range profile.DeviceResources {
+		resources[i] = models.DeviceResource{
+			Description: r.Description,
+			Name:        r.Name,
+			IsHidden:    isResourceHidden(r.Name, profile),
+			Tags:        toV2Tags(r.Tags),
+			Properties: models.ResourceProperties{
+				ValueType:    strings.Title(strings.ToLower(r.Properties.Value.Type)),
+				ReadWrite:    r.Properties.Value.ReadWrite,
+				Units:        r.Properties.Units.DefaultValue,
+				Minimum:      r.Properties.Value.Minimum,
+				Maximum:      r.Properties.Value.Maximum,
+				DefaultValue: r.Properties.Value.DefaultValue,
+				Mask:         r.Properties.Value.Mask,
+				Shift:        r.Properties.Value.Shift,
+				Scale:        r.Properties.Value.Scale,
+				Offset:       r.Properties.Value.Offset,
+				Base:         r.Properties.Value.Base,
+				Assertion:    r.Properties.Value.Assertion,
+				MediaType:    r.Properties.Value.MediaType,
+			},
+			Attributes: toV2Attributes(r.Attributes),
+		}
+	}
+
+	return resources
+}
+
+func isResourceHidden(resourceName string, profile v1Models.DeviceProfile) bool {
+	// Check whether the resource exists in the CoreCommands, if exists, the resource is not hidden.
+	for _, coreCommand := range profile.CoreCommands {
+		for _, res := range coreCommand.Get.Responses {
+			isContains := contains(res.ExpectedValues, resourceName)
+			if isContains {
+				return false
+			}
+		}
+		isContains := contains(coreCommand.Put.ParameterNames, resourceName)
+		if isContains {
+			return false
+		}
+	}
+	// Check whether the resource exists in the DeviceCommands, if exists, the resource is not hidden.
+	for _, deviceCommand := range profile.DeviceCommands {
+		for _, ro := range deviceCommand.Get {
+			if ro.DeviceResource == resourceName {
+				return false
+			}
+		}
+		for _, ro := range deviceCommand.Set {
+			if ro.DeviceResource == resourceName {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func contains(list []string, e string) bool {
+	for _, v := range list {
+		if v == e {
+			return true
+		}
+	}
+	return false
+}
+
+func toV2Tags(tags map[string]string) map[string]interface{} {
+	dto := make(map[string]interface{})
+	for k, v := range tags {
+		dto[k] = v
+	}
+	return dto
+}
+
+func toV2Attributes(attributes map[string]string) map[string]interface{} {
+	dto := make(map[string]interface{})
+	for k, v := range attributes {
+		dto[k] = v
+	}
+	return dto
+}
+
+func toV2DeviceCommands(deviceCommands []v1Models.ProfileResource) []models.DeviceCommand {
+	commands := make([]models.DeviceCommand, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = models.DeviceCommand{
+			Name:               c.Name,
+			IsHidden:           false,
+			ReadWrite:          "",
+			ResourceOperations: nil,
+		}
+		var ros []models.ResourceOperation
+		if len(c.Get) > 0 && len(c.Set) > 0 {
+			commands[i].ReadWrite = common.ReadWrite_RW
+			for _, getOp := range c.Get {
+				for _, setOp := range c.Set {
```

Comment:
check whether the Get and Set resource operations are the same.  if yes, just copy the get resource operation.
if no, copy the get resource operations with readwrite = R, and add an additional DeviceCommand named "{original name}_Set" with readwrite =W and copy the set resource operations.

Reply:
Modified as you suggest.